### PR TITLE
Move flash errors to service

### PR DESCRIPTION
**Pull request by locks**
## Tasks

~- [ ] Write automated tests to check flash errors behaviours and flows~ (moved to https://github.com/rust-lang/crates.io/issues/827)

## QA

- Make sure `set('flashError', …)` was replaced with `show(…)`
- Make sure `set('nextFlashError', …)` was replaced with `queue(…)`
- Manually test all of the flash error situations

### CODE REVIEW

File: app/controllers/me/index.js [hunk 1]
```diff
@@ -21,7 +21,7 @@ export default Ember.Controller.extend({
                 } else {
                     msg = 'An unknown error occurred';
                 }
-                this.controllerFor('application').set('nextFlashError', msg);
+                this.get('flashMessages').queue(msg);
```
Comment (Turbo87) [inline app/controllers/me/index.js]: missing service injection?

File: app/services/crate-search.js [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import Ember from 'ember';
+
+export default Ember.Service.extend({
+});
```
Comment (Turbo87) [inline app/services/crate-search.js]: seems unrelated?

File: app/templates/components/flash-message.hbs [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{{#if hasBlock}}
+  {{yield message}}
+{{else}}
```
Comment (Turbo87) [inline app/templates/components/flash-message.hbs]: do we need the `hasBlock` condition? seems unnecessary

File: app/controllers/application.js [hunk 4]
```diff
@@ -5,8 +5,7 @@ const { observer } = Ember;
 export default Ember.Controller.extend({
     searchController: Ember.inject.controller('search'),
 
-    flashError: null,
-    nextFlashError: null,
+    flashMessages: Ember.inject.service(),
```
Comment (Turbo87) [inline app/controllers/application.js]: injection seems unused?

File: app/templates/components/flash-message.hbs (hunk 3 above)
Comment (locks) [inline app/templates/components/flash-message.hbs]: 👼 I need a bit of whimsy

**Comment by locks** [comment]
@Turbo87 had some weirdness with the commits and GitHub, should be fixed

**Comment by Turbo87** [comment]
@locks one open comment and CI is apparently failing
